docs/src/main/asciidoc/kafka-schema-registry-avro.adoc

```diff
@@ -0,0 +1,552 @@
+  http://localhost:8080/movies
+
+curl --header "Content-Type: application/json" \
+  --request POST \
+  --data '{"title":"The Godfather","year":1972}' \
+  http://localhost:8080/movies
+
+curl --header "Content-Type: application/json" \
+  --request POST \
+  --data '{"title":"The Dark Knight","year":2008}' \
+  http://localhost:8080/movies
+
+curl --header "Content-Type: application/json" \
+  --request POST \
+  --data '{"title":"12 Angry Men","year":1957}' \
+  http://localhost:8080/movies
+----
+
+Observe what is printed in the second terminal. You should see something along the lines of:
+[source]
+----
+data:'The Shawshank Redemption' from 1994
+
+data:'The Godfather' from 1972
+
+data:'The Dark Knight' from 2008
+
+data:'12 Angry Men' from 1957
+----
+
+
+== Testing the application
+
+=== Infrastructure for tests
+We will now use Testcontainers to set up Kafka and Apicurio Schema Registry for tests.
+
+First, let's add test dependencies on Awaitility, Testcontainers and Strimzi to `pom.xml`:
+[source,xml]
+----
+<dependencies>
+    ...
+    <dependency>
+        <groupId>org.testcontainers</groupId>
+        <artifactId>testcontainers</artifactId>
+        <scope>test</scope>
+    </dependency>
+    <dependency>
+        <groupId>io.strimzi</groupId>
+        <artifactId>strimzi-test-container</artifactId>
+        <version>0.22.1</version>
+        <scope>test</scope>
+        <exclusions>
+            <exclusion>
+                <groupId>org.apache.logging.log4j</groupId>
+                <artifactId>log4j-core</artifactId>
+            </exclusion>
+        </exclusions>
+    </dependency>
+    <dependency>
+        <groupId>org.awaitility</groupId>
+        <artifactId>awaitility</artifactId>
+        <scope>test</scope>
+    </dependency>
+</dependencies>
+----
+
+Now, let's define a link:getting-started-testing.adoc#quarkus-test-resource[QuarkusTestResourceLifecycleManager] that will create the appropriate
+containers:
+[source,java]
+----
+package org.acme.kafka;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.testcontainers.containers.GenericContainer;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import io.strimzi.StrimziKafkaContainer;
+
+public class KafkaAndSchemaRegistryTestResource implements QuarkusTestResourceLifecycleManager {
+
+    private static final StrimziKafkaContainer kafka = new StrimziKafkaContainer();
+
+    private static GenericContainer<?> registry;
+
+    @Override
+    public Map<String, String> start() {
+        kafka.start();
+        registry = new GenericContainer<>("apicurio/apicurio-registry-mem:1.2.2.Final")
+                .withExposedPorts(8080)
+                .withEnv("QUARKUS_PROFILE", "prod")
+                .withEnv("KAFKA_BOOTSTRAP_SERVERS", kafka.getBootstrapServers())
+                .withEnv("APPLICATION_ID", "registry_id")
+                .withEnv("APPLICATION_SERVER", "localhost:9000");
```

Comment:
Could this instantiation be also inlined in the `registry` field declaration, just like `kafka = new StrimziKafkaContainer`?

Reply:
kafka.getBootstrapServers() requires the mapped port for Kafka, so I don't think it can be used before kafka.start()

Reply:
Ah sorry, didn't notice that one!